### PR TITLE
Resolve the `basePath` before constructing the `targetDir`

### PR DESCRIPTION
**Pull request by benjdlambert**

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]

**Review by freben — APPROVED** [review]
Alright. Is there anywhere else that might have been similarly affected?

**Comment by benjdlambert** [comment]
@freben not sure - but this was the only change from #23180. Just gonna do a whole release again I think - easier that way, and releases are pretty cheap.

File: packages/backend-common/src/paths.ts [hunk 1]
```diff
@@ -63,15 +63,17 @@ export function resolvePackagePath(name: string, ...paths: string[]) {
...
-  if (!isChildPath(resolveRealPath(base), resolveRealPath(targetPath))) {
+  if (!isChildPath(resolvedBasePath, resolveRealPath(targetPath))) {
     throw new NotAllowedError(
       'Relative path is not allowed to refer to a directory outside its parent',
     );
   }
 
-  return targetPath;
+  // Don't return the resolved path as the original could be a symlink
+  return resolvePath(base, path);
```
Comment (Rugvip) [inline packages/backend-common/src/paths.ts:76]: Yep think this is the way to go

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**